subprojects/core-api/src/main/java/org/gradle/api/artifacts/ComponentMetadataDetails.java

```diff
@@ -52,4 +54,40 @@
      * @param statusScheme the status scheme of the component
      */
     void setStatusScheme(List<String> statusScheme);
+
+    /**
+     * Remove a dependency from the component.
+     *
+     * @param dependencyNotation the dependency to remove (e.g. 'org.example:module1')
+     * @since 4.4
+     */
+    void removeDependency(Object dependencyNotation);
```

Comment:
If we directly exposed the dependency list, the user could do `dependencies.removeAll { /* any desired matching logic */ }` instead of having to specify the exact notation.